sentry-android-core/src/main/java/io/sentry/android/core/SentryStartupProfilingProvider.java

```diff
@@ -0,0 +1,147 @@
+    this.logger = logger;
+    this.buildInfoProvider = buildInfoProvider;
+  }
+
+  public SentryStartupProfilingProvider() {
+    logger = new AndroidLogger();
+    buildInfoProvider = new BuildInfoProvider(logger);
+  }
+
+  @Override
+  public boolean onCreate() {
+    final @Nullable Context context = getContext();
+
+    if (context == null) {
+      logger.log(SentryLevel.FATAL, "App. Context from ContentProvider is null");
+      return false;
+    }
+
+    // Debug.startMethodTracingSampling() is only available since Lollipop
+    if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.LOLLIPOP) {
+      return false;
+    }
+
+    final @NotNull File cacheDir = AndroidOptionsInitializer.getCacheDir(context);
+    final @NotNull File configFile = new File(cacheDir, STARTUP_PROFILING_CONFIG_FILE_NAME);
+
+    // No config exists: startup profiling is not enabled
+    if (!configFile.exists() || !configFile.canRead()) {
+      return false;
+    }
+
+    try (final @NotNull Reader fileReader = new SentryFileReader(configFile)) {
+      final @Nullable SentryStartupProfilingOptions profilingOptions =
+          new JsonSerializer(SentryOptions.empty())
+              .deserialize(fileReader, SentryStartupProfilingOptions.class);
+
+      if (profilingOptions == null) {
+        logger.log(
+            SentryLevel.WARNING,
+            "Unable to deserialize the SentryStartupProfilingOptions. Startup profiling will not start.");
+        return false;
+      }
+
+      if (!profilingOptions.isProfilingEnabled()) {
+        logger.log(SentryLevel.INFO, "Profiling is not enabled. Startup profiling will not start.");
+        return false;
+      }
+
+      final @NotNull TracesSamplingDecision startupSamplingDecision =
+          new TracesSamplingDecision(
+              profilingOptions.isTraceSampled(),
+              profilingOptions.getTraceSampleRate(),
+              profilingOptions.isProfileSampled(),
+              profilingOptions.getProfileSampleRate());
+      // We store any sampling decision, so we can respect it when the first transaction starts
+      AppStartMetrics.getInstance().setStartupSamplingDecision(startupSamplingDecision);
+
+      if (!(startupSamplingDecision.getProfileSampled() && startupSamplingDecision.getSampled())) {
+        return false;
+      }
+      logger.log(SentryLevel.DEBUG, "Startup profiling started.");
+
+      final @NotNull ITransactionProfiler startupProfiler =
+          new AndroidTransactionProfiler(
+              context.getApplicationContext(),
+              buildInfoProvider,
+              new SentryFrameMetricsCollector(
+                  context.getApplicationContext(), logger, buildInfoProvider),
+              logger,
+              profilingOptions.getProfilingTracesDirPath(),
+              profilingOptions.isProfilingEnabled(),
+              profilingOptions.getProfilingTracesHz(),
+              new SentryExecutorService());
```

Comment:
so this means we'll have 2 SentryExecutorServices now when startup profiling is enabled, right? I guess this should be fine, we'll have 2 threads with the same name, but that shouldn't be a problem https://stackoverflow.com/a/4978149

Reply:
yeah, we'd have a new thread for the profiler only